packages/wallets/particle/src/adapter.ts

```diff
@@ -0,0 +1,185 @@
+
+    private _connecting: boolean;
+    private _wallet: SolanaWallet | null;
+    private _publicKey: PublicKey | null;
+    private _config: ParticleAdapterConfig;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+
+    constructor(config: ParticleAdapterConfig) {
```

Comment:
This can be optional, e.g. https://github.com/solana-labs/wallet-adapter/blob/4d95a46f7cabdf423123d029db55667923df972e/packages/wallets/phantom/src/adapter.ts#L68